src/Foundation/NSObject2.cs

```diff
@@ -428,7 +430,8 @@ void CreateManagedRef (bool retain)
 		void ReleaseManagedRef ()
 		{
 			var handle = this.Handle; // Get a copy of the handle, because it will be cleared out when calling Runtime.NativeObjectHasDied, and we still need the handle later.
-			var user_type = Runtime.IsUserType (handle);
+			if (!Runtime.TryGetIsUserType (handle, out var user_type, out var error_message))
+				throw new InvalidOperationException ($"Unable to release the managed reference for the pointer {handle.ToString ()} whose managed type is {GetType ().FullName} because it wasn't possible to get the class of the pointer: {error_message}");
```

Comment:
```suggestion
				throw new InvalidOperationException ($"Unable to release the managed reference for the pointer {handle} whose managed type is {GetType ().FullName} because it wasn't possible to get the class of the pointer: {error_message}");
```
Same question